supplementary/alica_ros1/alica_ros_turtlesim/libalica-turtlesim/src/FourCorners.cpp

```diff
@@ -0,0 +1,51 @@
+#include "FourCorners.h"
+#include "FourCornersSummand.h"
+#include "engine/USummand.h"
+#include "engine/blackboard/Blackboard.h"
+#include "ros/ros.h"
+namespace turtlesim
+{
+
+std::shared_ptr<FourCornersUtilityFunction> FourCornersUtilityFunction::create(alica::UtilityFunctionContext& context)
+{
+    return std::make_shared<FourCornersUtilityFunction>(context);
+}
+
+/*
+    The utility function combines three components to determine the best assignment:
+    1. Priority (prio): A measure of how important the assignment is considering agent role and it's priority to take up a task.
+    2. Similarity (sim): A measure of how close the new assignment is to the previous assignment.
+    3. Custom component (summand): A user-defined component to add extra criteria.
+
+    Utility = priorityWeight * prio(assignment) + similarityWeight * sim(assignment, oldAssignment) + customWeight * summand(assignment)
```

Comment:
if the weights are set from the start, why are we showing this formula here without the weight values and with priority stuff which isn't used? Later on, we don't mention prio anywhere exactly because its weight is 0

Reply:
- I want to have the whole generic equation of utilty function here and refering the exact weight values we are using as it helps understand overall utilty concept better. 
- Even though priority term is not used still person refering this tutorial  should atleast know the significance of it in the whole equation, as it is one of the paramater of utility function.